File: app/src/main/java/com/platdmit/simplecloudmanager/app/adapters/ServerListAdapter.java
package com.platdmit.simplecloudmanager.app.adapters;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.platdmit.simplecloudmanager.R;
import com.platdmit.simplecloudmanager.domain.models.Server;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import static androidx.navigation.Navigation.findNavController;

public class ServerListAdapter extends RecyclerView.Adapter<ServerListAdapter.ServerListHolder> {

    private List<Server> elementList;

    @NonNull
    @Override
    public ServerListHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        int layoutType = R.layout.fragment_servers_item;
        return new ServerListHolder(layoutInflater, parent, layoutType);
    }

    @Override
    public void onBindViewHolder(@NonNull ServerListHolder holder, int position) {
        holder.bindData(elementList.get(position));
    }

    @Override
    public int getItemCount() {
        return elementList.size();
    }

    public void setContentData(List<Server> elements){
        this.elementList = elements;
    }

    class ServerListHolder extends RecyclerView.ViewHolder{

        private ImageView mImageLogo = itemView.findViewById(R.id.server_image_logo);

        private TextView mName = itemView.findViewById(R.id.server_name);;
        private TextView mUptime = itemView.findViewById(R.id.server_uptime);
        private TextView mId = itemView.findViewById(R.id.server_id);
        private TextView mStatus = itemView.findViewById(R.id.server_status);

        public ServerListHolder(LayoutInflater inflater, ViewGroup parent, int layoutType) {
            super(inflater.inflate(layoutType, parent, false));
        }

        public void bindData(Server data){
            mName.setText(data.getName());
            mUptime.setText(data.getUptime());
            mId.setText(String.valueOf(data.getId()));
            mStatus.setText(data.getStatus());

            itemView.setOnClickListener(v -> {
                Bundle bundle = new Bundle();
                bundle.putLong("ELEMENT_ID", data.getId());
                findNavController(v).navigate(R.id.serverFragment, bundle);
            });
        }
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/adapters/ServerListAdapter.kt
package com.platdmit.simplecloudmanager.adapters

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.os.bundleOf
import androidx.navigation.Navigation
import androidx.recyclerview.widget.RecyclerView
import com.platdmit.simplecloudmanager.R
import com.platdmit.simplecloudmanager.adapters.ServerListAdapter.ServerListHolder
import com.platdmit.domain.models.Server

class ServerListAdapter : RecyclerView.Adapter<ServerListHolder>() {
    private lateinit var elementList: List<Server>

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ServerListHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val layoutType = R.layout.fragment_servers_item
        return ServerListHolder(layoutInflater, parent, layoutType)
    }

    override fun onBindViewHolder(holder: ServerListHolder, position: Int) {
        holder.bindData(elementList[position])
    }

    override fun getItemCount(): Int {
        return elementList.size
    }

    fun setContentData(elements: List<Server>) {
        elementList = elements
    }

    inner class ServerListHolder(inflater: LayoutInflater, parent: ViewGroup?, layoutType: Int) : RecyclerView.ViewHolder(inflater.inflate(layoutType, parent, false)) {
        private val mImageLogo = itemView.findViewById<ImageView>(R.id.server_image_logo)
        private val mName = itemView.findViewById<TextView>(R.id.server_name)
        private val mUptime = itemView.findViewById<TextView>(R.id.server_uptime)
        private val mId = itemView.findViewById<TextView>(R.id.server_id)
        private val mStatus = itemView.findViewById<TextView>(R.id.server_status)
        fun bindData(data: Server) {
            mName.text = data.name
            mUptime.text = data.uptime
            mId.text = data.id.toString()
            mStatus.text = data.status
            itemView.setOnClickListener {
                Navigation.findNavController(it).navigate(R.id.serverFragment, bundleOf("ELEMENT_ID" to data.id))
            }
        }
    }
}